open-codegen/opengen/tcp/solver_response.py

```diff
@@ -0,0 +1,36 @@
+from .solver_status import SolverStatus
+from .solver_error import SolverError
+
+class SolverResponse:
+    """Stores a solver response of type SolverStatus or SolverError."""
+    def __init__(self, d):
+        """Constructs instance of <code>SolverResponse</code>
+
+        Args:
+            d: dictionary containing either status or error attributes
+
+        Returns:
+            New instance of <code>SolverResponse</code>
+        """
+        if 'Error' in d.values():
+            self.response = SolverError(d)
+        else:
+            self.response = SolverStatus(d)
+
+    def is_ok(self):
+        """Determines if response is OK."""
+        return isinstance(self.response, SolverStatus)
+
+    def is_err(self):
```

Comment:
Since `is_err()` is True if and only if `is_ok()` is False, you can remove this method